src/bdk.udl

```diff
@@ -1,9 +1,23 @@
 namespace bdk {
   [Throws=BdkError]
-  ExtendedKeyInfo generate_extended_key(Network network, WordCount word_count, string? password);
+  string generate_mnemonic(WordCount word_count);
+};
+
+interface DerivationPath {
+  [Throws=BdkError]
+  constructor(string path);
+};
 
+interface ExtendedPrivKey {
   [Throws=BdkError]
-  ExtendedKeyInfo restore_extended_key(Network network, string mnemonic, string? password);
+  constructor(Network network, string mnemonic, string? password);
+  ExtendedPrivKey derive_xprv(DerivationPath? derivation_path);
+  ExtendedPubKey derive_xpub(DerivationPath? derivation_path);
+  string to_string();
+};
+
+interface ExtendedPubKey {
+  string to_string();
```

Comment:
Looks like this won't compile in Kotlin since `toString()` is an open, member function of `Any` that needs to be overriden and uniffi doesn't seem to have a way to declare an overriding API, as far as I can tell. A simple workaround here would be to rename `to_string()` to something else (maybe `ExtendedPubKey#value` or `#raw` read-only property?)

Reply:
Filed an issue with uniffi to see if it's feasible to allow adding `override` modifier in Kotlin: https://github.com/mozilla/uniffi-rs/issues/1286.